osgp/platform/osgp-adapter-domain-publiclighting/src/main/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/tasks/DeviceConnection104LmdScheduledTask.java

```diff
@@ -30,22 +41,65 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(DeviceConnection104LmdScheduledTask.class);
 
-    private static final String PROTOCOL = "60870-5-104";
-
     @Autowired
     private int deviceConnection104LmdScheduledTaskMaximumAllowedAge;
 
     @Override
     public void run() {
         LOGGER.info("Ensuring active connections with IEC 60870-5-104 light measurement devices");
         try {
-            final List<LightMeasurementDevice> devices = this.findLightMeasurementDevicesByProtocol(PROTOCOL);
-            final List<LightMeasurementDevice> devicesToConnect = this.findLightMeasurementDevicesToConnect(devices,
-                    this.deviceConnection104LmdScheduledTaskMaximumAllowedAge);
-            this.connectLightMeasurementDevices(devicesToConnect);
+            final List<RtuDevice> gatewaysToConnect = this
+                    .findLightMeasurementGateways(this.deviceConnection104LmdScheduledTaskMaximumAllowedAge);
+            this.connectLightMeasurementGateways(gatewaysToConnect);
+
         } catch (final Exception e) {
             LOGGER.error("Exception caught ensuring active connection to IEC 60870-5-104 light measurement devices", e);
         }
     }
 
+    protected List<RtuDevice> findLightMeasurementGateways(final int maxAllowedAge) {
+        LOGGER.info("Trying to find light measurement gateways");
+
+        final RtuDeviceSpecifications specifications = new RtuDeviceSpecifications();
+
+        final Specification<RtuDevice> specification = specifications
+                .hasDeviceLifeCycleStatus(DeviceLifecycleStatus.IN_USE)
+                .and(specifications.hasDeviceType("LMG"))
+                .and(specifications
+                        .hasLastCommunicationTimeBefore(Instant.now().minus(maxAllowedAge, ChronoUnit.HOURS)));
```

Comment:
I don't think it is needed to check for empty values, as it is default set to `Instant.EPOCH` in the `RtuDevice` class?

Reply:
For the current state of the world, it looks like you are absolutely right.
Where we appear to prefer different approaches is in the level of defensiveness in the programming.
Taking into account what to do if the last communication time would not be set, is only a little bit of extra work.
If someone later-on decides that a default value in the last communication time does not make sense, and null would be better, the code would no longer work. This is a risk, perhaps a small one, that I would rather avoid.
The amount of time it would take to find out this code is tied to the data not being null and fixing the data or the code, would soon be greater than the time it takes to program defensively against this.
As I don't think our coding standards say anything about the level of defensiveness, you can leave it as is if you are not convinced by my arguments.

Reply:
No longer relevant, code has been removed.